app/models/concerns/has_phone_number_concern.rb

```diff
@@ -4,45 +4,28 @@ module HasPhoneNumberConcern
   extend ActiveSupport::Concern
 
   included do
-    before_validation :format_phone_number
-    validates :phone_number, length: {minimum: 7, maximum: 15, allow_blank: false}
+    before_validation :format_phone_number, if: :phone_number?
+    validates :phone_number, phone: {
+      types: %i[mobile],
+      allow_blank: false
+    }
   end
 
   def human_friendly_phone_number
-    phone_number.gsub(/^33/, "0").match(/(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})/)&.captures&.join(" ")
+    if parsed_phone_number.valid_for_country?(:fr)
+      parsed_phone_number.national
+    else
+      parsed_phone_number.e164
```

Comment:
thoughts for always using e164?

Reply:
The phone number is always in e164 in the database with this PR. This instance method is just a convenient way to display the phone number in the view for the user.
Your point's valid though. But it's more a feature debate that's worth discussing in the linked issue in my opinion

Reply:
makes sense to display the national version of it for french numbers.